test/parallel/test-validators.js

```diff
@@ -85,3 +89,35 @@ const invalidArgValueError = {
 
   validateObject(null, 'foo', { nullable: true });
 }
+
+{
+  // validateString tests
+  validateString('hi', 'foo');
+
+  [1, false, [], {}, NaN, 1n, null, undefined].forEach((i) => {
+    assert.throws(() => validateString(i, 'foo'), invalidArgTypeError);
+  });
+}
+
+{
+  // validateBuffer tests
+  validateBuffer(Buffer.from('hi'), 'foo');
+  validateBuffer(Buffer.alloc(10), 'foo');
+  validateBuffer(new Uint8Array(10), 'foo');
+  validateBuffer(new DataView((new Uint8Array(10)).buffer));
+
+  [1, false, '', {}, [], 1n, null, undefined].forEach((i) => {
+    assert.throws(() => validateBuffer(i, 'foo'), invalidArgTypeError);
+  });
```

Comment:
Do we need these tests? I suggest to add more "integration" tests instead that test these functions where the code is actually implemented. That way we are certain that the code is properly executed by looking at the coverage report.